.github/workflows/cli.yaml

```diff
@@ -0,0 +1,134 @@
+name: Karmada CLI Workflow
+on:
+  # Run this workflow every time a new commit pushed to upstream/fork repository.
+  # Run workflow on fork repository will help contributors find and resolve issues before sending a PR.
+  push:
+  pull_request:
+jobs:
+  codegen:
+    name: codegen
+    runs-on: ubuntu-20.04
+    env:
+      GOPATH: ${{ github.workspace }}
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/karmada-io/karmada
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/karmada-io/karmada
+      - name: install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.5
+      - name: Install Protoc
+        # TODO(@RainbowMango): Update the action version to adopt Node16
+        # track issue: https://github.com/arduino/setup-protoc/issues/59
+        uses: arduino/setup-protoc@v1
+        with:
+          version: '3.x'
+          # Use the automatic token, so that this task can be run in the forked repo.
+          # https://docs.github.com/en/actions/security-guides/automatic-token-authentication
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: verify codegen
+        run: hack/verify-codegen.sh
+      - name: verify crdgen
+        run: hack/verify-crdgen.sh
+      - name: verify protobuf
+        run: hack/verify-estimator-protobuf.sh
+      - name: verify swagger docs
+        run: hack/verify-swagger-docs.sh
+      - name: verify lifted docs
+        run: hack/verify-lifted.sh
+  build:
+    name: compile
+    needs: codegen # rely on codegen successful completion
+    runs-on: ubuntu-20.04
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+        with:
+          # Number of commits to fetch. 0 indicates all history for all branches and tags.
+          # We need to guess version via git tags.
+          fetch-depth: 0
+      - name: install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.5
+      - name: compile
+        run: |
+          set -x
+
+          # tar the newest crds
+          cd  charts/karmada/
+          cp -r _crds crds
+          tar -zcvf ../../crds.tar.gz crds
+          cd -
+
+          # prepare karmadactl and images
+          make karmadactl
+          cp _output/bin/linux/amd64/karmadactl /usr/local/bin/
+          export VERSION=ci-e2e
+          make images
+  karmadactlinit:
+    name: karmadactl init installation test
+    needs: build
```

Comment:
Please check if we can [define a prerequisite job](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#defining-prerequisite-jobs) from another workflow.

Reply:
I checked, but couldn't.